### PR TITLE
chore: make package ready for packagist

### PR DESCRIPTION
**Pull request by Pascalmh**

### CODE REVIEW

**Comment by Pascalmh** [comment]
The vendor folder is checked in, that might be a problem… Would you consider managing dependencies through submodules/composer instead of adding the vendor-folder into your repo (which would be [best practice](https://getcomposer.org/doc/faqs/should-i-commit-the-dependencies-in-my-vendor-directory.md))?

**Comment by Pascalmh** [comment]
it seems like the install instructions you provide do not work anyway - https://github.com/S1SYPHOS/kirby-webp/tree/master/vendor/rosell-dk <- this dependency seems to be missing

**Comment by S1SYPHOS** [comment]
It's fixed now, for general Composer discussion, see your other PR!

**Comment by S1SYPHOS** [comment]
Sorry for closing this, thought I was on #3 :man_dancing: 

Thanks for mentioning the installation didn't work btw - now, regarding Composer, I [opened a poll](https://forum.getkirby.com/t/favorite-mechanism-for-including-plugins/9818) in the Kirby forum, since it's a quite interesting topic - wether if people are using Git, Composer, simply downloading stuff (or even using Kirby CLI), feel free to join!

Here's the thing: while I like what Composer does for the PHP world, Git submodules are just as effective for Kirby purposes (since Kirby itself is incompatible with Composer). This might change in the future, and I'm open to releasing all my plugins to Packagist as soon as this lands! Because of that, I want to keep this PR open, and be assured that I'm grateful for your interest in `kirby-webp`.

Let me know what you think!

Cheers,
S1SYPHOS

**Comment by S1SYPHOS** [comment]
Update:
As far as I know, releasing it as Composer plugin does not rely on the master branch, so that is an easy way to make everyone happy, no?

File: composer.json [hunk 1]
```diff
@@ -1,11 +1,14 @@
 {
+  "name": "S1SYPHOS/kirby-webp",
+  "type": "kirby-plugin",
```
Comment (S1SYPHOS) [inline composer.json:3]: If I understand correctly, this maps the install location to `site/plugins/`, doesn't it? My projects tend to have a different structure, inside root there are `source`, `build`, `package.json` and so on .. so basically the baseDir, the directory containing `site`, would be `build`.

How will this be configured? From the issues already raised about this, I get it that in this case you have to override it. Are you familiar with this? Maybe there's an easier way :+1: 

// Edit:
Merging your PR doesn't depend on this question in any way, that's just for my personal information need ..

**Review by S1SYPHOS — CHANGES_REQUESTED** [review]
Please switch your PR to the `composer` branch & adjust your code accordingly.

Then I'll be happy to merge!

File: composer.json (hunk 1 above)
Comment (Pascalmh) [reply composer.json:3]: Yes it puts all files inside the site/plugins/your-plugin-name folder. I am not sure what you mean with the build and source folder but all your files should be inside a folder that's inside the sites/plugins folder...

Comment (S1SYPHOS) [reply composer.json:3]: Nevermind

**Comment by S1SYPHOS** [comment]
Changed branch name in `composer.json` & deleted `vendor`, so this should be good now.
Thanks again for your effort!

**Pull request merged**